Python/bltinmodule.c

```diff
@@ -1928,10 +1928,12 @@ builtin_input_impl(PyObject *module, PyObject *prompt)
 
         stdin_encoding = _PyObject_GetAttrId(fin, &PyId_encoding);
         stdin_errors = _PyObject_GetAttrId(fin, &PyId_errors);
-        if (!stdin_encoding || !stdin_errors)
+        if (!stdin_encoding || !stdin_errors ||
+            !PyUnicode_Check(stdin_encoding) ||
+            !PyUnicode_Check(stdin_errors))
```

Comment:
This is the main point of this change. If `PyUnicode_AsUTF8()` is failed due to memory error or unpaired surrogates, this is an error. But if *encoding* or *error* are not strings (but `None` for example), use the fallback.